### ISSUE TITLE
Not opening on macOS 14.4.1

### ISSUE THREAD

**Issue by skhaz** [alacritty/alacritty#7919]
I have installed it using brew (downloading from the website results in the same issue). When I open it, it closes immediately.

I tried to debug it on the command line, and here is what I got.

```shell
λ /Applications/ alacritty
[1]    27904 killed     alacritty
λ /Applications/ lldb alacritty
(lldb) target create "alacritty"
Current executable set to '/opt/homebrew/bin/alacritty' (arm64).
(lldb) r
Process 27928 launched: '/opt/homebrew/bin/alacritty' (arm64)
Process 27928 exited with status = 9 (0x00000009) Terminated due to signal 9
```

**Comment by chrisduerr** [comment]
Alacritty is getting killed externally. Two possibilities I can think of is either the shell trying to kill a misbehaving child process, or an issue with macOS' permissions system.

If you run Alacritty with `-vvv` to get the maximum debug output, do you get *anything*? If not, this is probably not an Alacritty issue.

**Comment by skhaz** [comment]
No luck

```
alacritty -vvv
[1]    31225 killed     alacritty -vvv
```

**Comment by skhaz** [comment]
Other apps, like iterm2 works fine.

**Comment by chrisduerr** [comment]
Do you get any notification when opening it through macOS' UI? Is there an option to right click -> open in the file manager?

Can also try https://github.com/alacritty/alacritty/wiki/Self-Signing-Alacritty-on-macOS.

**Comment by skhaz** [comment]
No notifications. I am a experienced macOS user (since 2009), there is something strange. Tomorrow I will dig a bit deeper.

**Comment by slurpyb** [comment]
I just installed 14.4.1 and installed Alacritty for the first time on this machine (I migrated from Ubuntu) and I am able to use the application after going into Privacy and Security and adding an exception for Alacritty.

**Comment by ninthsun91** [comment]
> I just installed 14.4.1 and installed Alacritty for the first time on this machine (I migrated from Ubuntu) and I am able to use the application after going into Privacy and Security and adding an exception for Alacritty.
> 
> I used the release binary from here, not homebrew.

This worked for me, too.
![image](https://github.com/alacritty/alacritty/assets/50778955/eada408a-8eba-41b6-8b05-e8026cd00f57)

After a few times of denial from the system, now I can open Alacritty.

**Comment by b4lk0n** [comment]
I'm still experiencing this issue. Installed it via homebrew.
After opening Alacritty app, the terminal appears for a moment and closes right away.

This happens only when opening Alacritty from the Applications (spotlight/raycast).
When launching from another terminal, it just works well (`$ alacritty`).

Found the following logs in console:
```
2024-05-03 17:01:11.259637 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: internal event: WILL_SPAWN, code = 0
2024-05-03 17:01:11.259652 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: service state: spawn scheduled
2024-05-03 17:01:11.259654 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: service state: spawning
2024-05-03 17:01:11.259672 <Notice>: Coalition Cache Hit: app<application.org.alacritty.5092089.5092094(501)> [44301]
2024-05-03 17:01:11.259697 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: launching: launch job demand
2024-05-03 17:01:11.260838 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: xpcproxy spawned with pid 92843
2024-05-03 17:01:11.260862 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: internal event: SPAWNED, code = 0
2024-05-03 17:01:11.260865 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: service state: xpcproxy
2024-05-03 17:01:11.260910 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: internal event: SOURCE_ATTACH, code = 0
2024-05-03 17:01:11.267465 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: service state: running
2024-05-03 17:01:11.267477 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: internal event: INIT, code = 0
2024-05-03 17:01:11.267479 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: job state = running
2024-05-03 17:01:11.267557 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: Successfully spawned alacritty[92843] because launch job demand
2024-05-03 17:01:11.276034 (pid/92843 [alacritty]) <Notice>: uncorking exec source upfront
2024-05-03 17:01:11.276059 (pid/92843 [alacritty]) <Notice>: created
2024-05-03 17:01:11.419586 (pid/92843 [alacritty]) <Notice>: uncorking domain
2024-05-03 17:01:11.419790 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.ColorPickerImageLoader
2024-05-03 17:01:11.419799 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.defaultSyncService
2024-05-03 17:01:11.419801 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.documentPopoverViewService
2024-05-03 17:01:11.419804 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.LegacyExternalColorPickerService.arm64
2024-05-03 17:01:11.419806 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.LegacyExternalColorPickerService.x86_64
2024-05-03 17:01:11.419809 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.MediaLibraryBrowser
2024-05-03 17:01:11.419811 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.sandboxedServiceRunner
2024-05-03 17:01:11.419813 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.ThemeWidgetControlViewService
2024-05-03 17:01:11.419815 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.xpc.TypographyPanelService
2024-05-03 17:01:11.419818 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.automator.xpc.workflowServiceRunner
2024-05-03 17:01:11.419820 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.VersionsUIHelper
2024-05-03 17:01:11.419823 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.appkit.xpc.openAndSavePanelService
2024-05-03 17:01:11.419879 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.PerfPowerTelemetryClientRegistrationService
2024-05-03 17:01:11.419920 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.CoreGraphics.CGPDFService
2024-05-03 17:01:11.419965 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.foundation.UserScriptService
2024-05-03 17:01:11.419988 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.ImageIOXPCService
2024-05-03 17:01:11.420027 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.textkit.nsattributedstringagent
2024-05-03 17:01:11.420048 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.speech.SpeechSynthesisServerXPC
2024-05-03 17:01:11.420106 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.Core-Audio-Driver-Service.helper
2024-05-03 17:01:11.420108 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.Core-Audio-Driver-Service
2024-05-03 17:01:11.420109 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.DriverHelper
2024-05-03 17:01:11.420116 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.extensionkitservice
2024-05-03 17:01:11.420159 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.corelocation.maphelperservice
2024-05-03 17:01:11.420234 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.CoreDisplay.XPCService
2024-05-03 17:01:11.420241 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.CMValidateMovieDataReferenceService
2024-05-03 17:01:11.420276 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.AUHostingService.x86-64
2024-05-03 17:01:11.420279 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.AUHostingService.arm64e
2024-05-03 17:01:11.420281 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.AudioConverterService
2024-05-03 17:01:11.420285 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.CarbonComponentScanner.x86-64
2024-05-03 17:01:11.420289 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.ComponentTagHelper
2024-05-03 17:01:11.420292 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.InfoHelper
2024-05-03 17:01:11.420294 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.audio.SandboxHelper
2024-05-03 17:01:11.420302 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.internal.DataDetectorsRemoteScanner
2024-05-03 17:01:11.420324 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.OSLogService
2024-05-03 17:01:11.420411 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.coremedia.videodecoder
2024-05-03 17:01:11.420414 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.coremedia.videodecoder.zonto
2024-05-03 17:01:11.420416 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.coremedia.videoencoder
2024-05-03 17:01:11.420419 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.coremedia.videoencoder.zonto
2024-05-03 17:01:11.420530 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.ColorSyncXPCAgent
2024-05-03 17:01:11.420594 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.desktopservices.ArchiveService
2024-05-03 17:01:11.420598 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.desktopservices.QuarantineService
2024-05-03 17:01:11.420627 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.security.XPCAcmeService
2024-05-03 17:01:11.420676 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.MTLCompilerService
2024-05-03 17:01:11.420708 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.security.pboxd
2024-05-03 17:01:11.420754 (pid/92843 [alacritty]) <Notice>: Service stub created for com.apple.LoginUserService
2024-05-03 17:01:11.456634 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: exited due to exit(0)
2024-05-03 17:01:11.456648 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: service state: exited
2024-05-03 17:01:11.456660 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: internal event: EXITED, code = 0
2024-05-03 17:01:11.456663 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: job state = exited
2024-05-03 17:01:11.456692 (gui/501 [100021]) <Notice>: service inactive: application.org.alacritty.5092089.5092094
2024-05-03 17:01:11.456696 (gui/501/application.org.alacritty.5092089.5092094 [92843]) <Notice>: service state: not running
2024-05-03 17:01:11.457427 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: removing job: caller = runningboardd
2024-05-03 17:01:11.457441 (gui/501 [100021]) <Notice>: removing service: application.org.alacritty.5092089.5092094
2024-05-03 17:01:11.457450 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: internal event: PETRIFIED, code = 0
2024-05-03 17:01:11.457452 (gui/501/application.org.alacritty.5092089.5092094) <Notice>: job state = removed
2024-05-03 17:01:11.459185 (pid/92843 [alacritty]) <Notice>: shutting down
2024-05-03 17:01:11.459219 (pid/92843 [alacritty]) <Notice>: cleaning up
```

OS:
```
Software:

    System Software Overview:

      System Version: macOS 14.4.1 (23E224)
      Kernel Version: Darwin 23.4.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: 
      User Name: 
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 7 days, 2 hours, 47 minutes
```


```
$ alacritty --version
alacritty 0.13.2 (bb8ea18)
```

**Comment by b4lk0n** [comment]
So, as usual, identified the root cause right after commenting on the issue.

The issue was caused by a config file. I had set the `shell` to be `/bin/zsh` and this caused the app to crash when trying to launch from Finder.

Both `shell = "/bin/zsh"` and `shell = { program = "/bin/zsh" }` caused Alacritty to crash. And once I removed the `shell` option entirely, the terminal started working properly with `zsh`.

However, setting `shell = "/bin/bash"` works well. Alacritty launches and starts Bash.
